### ISSUE TITLE
Missing output for download-artifact@v3

### ISSUE THREAD

**Issue by ak-a** [rhysd/actionlint#442]
https://github.com/actions/download-artifact/tree/v3?tab=readme-ov-file#download-path-output

@v3 does have an download-path output as per documentation.  Can we have this added to the checks please?

https://github.com/rhysd/actionlint/blob/main/popular_actions.go#L611-L631
suggestion:
```
	"actions/download-artifact@v3": {
		Name: "Download a Build Artifact",
		Inputs: ActionMetadataInputs{
			"name": {"name", false},
			"path": {"path", false},
		},
		Outputs: ActionMetadataOutputs{
			"download-path": {"download-path"},
		},	},
	"actions/download-artifact@v4": {
		Name: "Download a Build Artifact",
		Inputs: ActionMetadataInputs{
			"github-token":   {"github-token", false},
			"merge-multiple": {"merge-multiple", false},
			"name":           {"name", false},
			"path":           {"path", false},
			"pattern":        {"pattern", false},
			"repository":     {"repository", false},
			"run-id":         {"run-id", false},
		},
		Outputs: ActionMetadataOutputs{
			"download-path": {"download-path"},
		},
```

**Comment by isaacsanders** [comment]
The output is set in the latest version of the code, it is just not declared in the `action.yml`

**Comment by rhysd** [comment]
This issue should be fixed by [v1.7.2](https://github.com/rhysd/actionlint/releases/tag/v1.7.2).

**Comment by isaacsanders** [comment]
Can you indicate where this was fixed in the diff?
https://github.com/rhysd/actionlint/compare/v1.7.1...v1.7.2 I couldn't find
a location.

**Comment by isaacsanders** [comment]
The relevant code
https://github.com/rhysd/actionlint/blob/v1.7.2/popular_actions.go#L613

**Comment by rhysd** [comment]
This commit: https://github.com/rhysd/actionlint/commit/51c772b355d285a239ea3da5761a6caf7332dfa5

Notice that diff is not expanded by default and you need to click 'Load diff'.

**Comment by isaacsanders** [comment]
The action mentioned in the issue about (`actions/download-artifact@v3`) was not updated. The problem with that action can’t be automatically fixed, it needs a specific override.

**Comment by rhysd** [comment]
Okay let me check it later.

**Comment by rhysd** [comment]
I checked the source. `actions/download-artifact@v3` doesn't define the output in its `action.yml` file so actionlint cannot know it. This is an issue of the action, not actionlint.

https://github.com/actions/download-artifact/blob/9bc31d5ccc31df68ecc42ccf4149144866c47d8a/action.yml

And I also confirmed that `actions/download-artifact@v4` defines the output well.

https://github.com/actions/download-artifact/blob/fa0a91b85d4f404e444e00e005971372dc801d16/action.yml#L36

So the simple way to fix this issue is just upgrading your dependency from v3 to v4.

**Comment by isaacsanders** [comment]
We are on GHES, so we can not upgrade to v4. v3 doesn't specify the output, but it does exist.

**Comment by isaacsanders** [comment]
I'm going to see how difficult it would be to contribute the change upstream.

**Comment by isaacsanders** [comment]
https://github.com/actions/download-artifact/issues/355

**Comment by rhysd** [comment]
Thanks for creating the issue. Meanwhile `-ignore` option would be a workaround for this.

**Comment by rhysd** [comment]
I added a workaround for this issue at d339f6e. It is shipped with the latest release v1.7.3.

https://github.com/rhysd/actionlint/releases/tag/v1.7.3

**Comment by isaacsanders** [comment]
You’re fantastic. I appreciate you.
